utils/utils.go

```diff
@@ -673,6 +686,15 @@ func CustomizeServiceAccount(sa *corev1.ServiceAccount, ba common.BaseComponent)
 	}
 }
 
+func removePullSecret(sa *corev1.ServiceAccount, pullSecretName string) {
+	for i, obj := range sa.ImagePullSecrets {
+		if obj.Name == pullSecretName {
+			sa.ImagePullSecrets = append(sa.ImagePullSecrets[:i], sa.ImagePullSecrets[i+1:]...)
```

Comment:
I am not entirely sure if a concurrent modification exception could happen in Go for this scenario (iterating through pull secrets and setting a value for the same). If we are not sure, getting the index of the pull secret we are looking for and setting this outside of for-loop is another option.